pkg/jobrunaggregator/jobrunaggregatorapi/types_row_historical_data.go

```diff
@@ -0,0 +1,95 @@
+package jobrunaggregatorapi
+
+import (
+	"encoding/json"
+	"fmt"
+)
+
+type HistoricalDataRow struct {
+	Name         string
+	Release      string
+	FromRelease  string
+	Platform     string
+	Architecture string
+	Network      string
+	Topology     string
+	P95          string
+	P99          string
+	Type         string
+}
+
+func (c *HistoricalDataRow) UnmarshalJSON(data []byte) error {
+	var values struct {
+		AlertName    string
+		BackendName  string
+		Release      string
+		FromRelease  string
+		Platform     string
+		Architecture string
+		Network      string
+		Topology     string
+		P95          string
+		P99          string
+	}
+	err := json.Unmarshal(data, &values)
+	if err != nil {
+		return err
+	}
+	c.Name = values.BackendName
+	c.Type = "disruption"
+	if values.AlertName != "" {
+		c.Name = values.AlertName
+		c.Type = "alert"
+	}
+	c.Release = values.Release
+	c.FromRelease = values.FromRelease
+	c.Platform = values.Platform
+	c.Architecture = values.Architecture
+	c.Network = values.Network
+	c.Topology = values.Topology
+	c.P95 = values.P95
+	c.P99 = values.P99
+	return nil
+}
+
+func (c *HistoricalDataRow) MarshalJSON() ([]byte, error) {
+	var values struct {
+		AlertName    string `json:",omitempty"`
+		BackendName  string `json:",omitempty"`
+		Release      string
+		FromRelease  string
+		Platform     string
+		Architecture string
+		Network      string
+		Topology     string
+		P95          string
+		P99          string
+	}
+	if c.Type == "alerts" {
+		values.AlertName = c.Name
+	} else {
+		values.BackendName = c.Name
+	}
+	values.Release = c.Release
+	values.FromRelease = c.FromRelease
+	values.Platform = c.Platform
+	values.Architecture = c.Architecture
+	values.Network = c.Network
+	values.Topology = c.Topology
+	values.P95 = c.P95
+	values.P99 = c.P99
+
+	return json.Marshal(values)
+}
+
+func (a *HistoricalDataRow) GetKey() string {
+	return fmt.Sprintf("%s_%s_%s_%s_%s_%s_%s",
```

Comment:
When a certain element (e.g. FromRelease) is empty, do we get double underscore? That might explain why this ordering is different from when I manually query with "Order By" with the same sequence of columns.

Reply:
Yup, that's what happens when they are empty